### ISSUE TITLE
replace fmincon for octave

### ISSUE THREAD

**Issue by tanguyduval** [qMRLab/qMRLab#74]
Gabriel feedback:
````
Information found on Octave compatibility function fmincon
Solutions:
1) function sqp(x0, phi) -> Not exactly the same as fmincon so might provoke errors if the code is not adapted
2)function nonlin_min (f, pin, settings) -> Same as before
3)download Optimization package for Octave forge with fmincon function
4) function npsol in octave

Continuing research
````

**Comment by tanguyduval** [comment]
I created a new issue for that so that we don't get mixed we other comments.
My comments:
  * The optimization package is already part of qMRLab. (installed in startup.m) You can use it if needed.
  * fmincon is used exclusively in all diffusion models. ex:
````
            if ~moxunit_util_platform_is_octave
                [xopt, residue] = fmincon(@(x) double(-2*sum(scd_model_likelihood_rician(datadif,max(eps,S0.*equation(obj, addfixparameters(obj.st,x,fixedparam))), SigmaNoise))), double(obj.st(~fixedparam)), [], [], [],[],double(obj.lb(~fixedparam)),double(obj.ub(~fixedparam)),[],optimoptions('fmincon','MaxIter',20,'display','off','DiffMinChange',0.03));
                obj.st(~fixedparam) = xopt; xopt = obj.st;
            end
````
  * You can compare validate the fitting of Octave using the Sim_Single_Voxel_Curve method. See `function test_sim` of [test_Models.m](https://github.com/neuropoly/qMRLab/blob/master/Test/MoxUnitCompatible/test_Models.m). Let me know when you think you have a good solution and I'll test that.

**Comment by agahkarakuzu** [comment]
Resolved #298.
